File: android/src/main/java/com/pplninja/plugins/capacitorplayaudiofromurl/PlayAudioFromUrlPlugin.java
package com.pplninja.plugins.capacitorplayaudiofromurl;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "PlayAudioFromUrl")
public class PlayAudioFromUrlPlugin extends Plugin {

    private PlayAudioFromUrl implementation = new PlayAudioFromUrl();

    @PluginMethod
    public void play(PluginCall call) {
        implementation.play(call);
    }
}

File: ios/Plugin/PlayAudioFromUrlPlugin.swift
import Foundation
import Capacitor
import AVFoundation

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(PlayAudioFromUrlPlugin)
public class PlayAudioFromUrlPlugin: CAPPlugin {
    private let implementation = PlayAudioFromUrl()

    @objc func play(_ call: CAPPluginCall) {
        let url = call.getString("url") ?? ""
        print("PlayAudioFromUrlPlugin.swift play: \(url)")
        implementation.play(url) { success, error in
            if success {
                call.resolve()
            } else {
                call.reject("play_error", error?.localizedDescription, error)
            }
        }
    }

//     @objc override public func checkPermissions(_ call: CAPPluginCall) {
//         // TODO
//     }
//
//     @objc override public func requestPermissions(_ call: CAPPluginCall) {
//         // TODO
//     }
}
